### ISSUE TITLE
[OT] Possible use of Tinythreadpp to replace C++11

### ISSUE THREAD

**Issue by Flix01** [pplux/px#6]
[I'm closing this soon because it's not a real issue]

I remember that some years ago I liked this library a lot: [[tinythreadpp]](http://tinythreadpp.bitsnbites.eu/) [[API reference here]](http://tinythreadpp.bitsnbites.eu/doc/).

From your docs, it seems that px_sched.h is:
* Portable, written in C++11 with no dependency
* C++11 only used for thread, mutex, and condition variable no STL container is used

whereas px_render.h is more complex (needs C++11/C++14).

I'm wondering if px_sched.h can be compiled without C++11, just using tinythreadpp.

AFAIR, the API is very similar (except a few missing constructor overloads in the thread class).

However, this post is just a useless experiment: nowadays most compilers support C++11.
Maybe this is just something for people trying to minimize file size (I don't think tinythreadpp uses STL, but I have to check).

So basically I'm going to close this now, leaving this post just as a suggestion for the few users that can be interested in removing the C++11 dependency from px_sched.h.

Sorry for polluting the Issue Section.

**Comment by Flix01** [comment]
On a closer look there all other dependencies to C++11 in px_sched.h:
* std::function (in ```#include <functional>```).
* ```nullptr``` (that can be replaced by NULL).
* ```#include <atomic>```

Then there are some lambdas in the examples.

So I guess I won't dig further into this task.
